### PR TITLE
feat: set Space state labels

### PR DESCRIPTION
**Pull request by MatousJobanek**
Set the Space state label values:
* `pending` when the `targetCluster` is NOT set
* `cluster-assigned` when the `targetCluster` IS set

paired PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/465

### CODE REVIEW

File: controllers/space/space_controller_test.go [hunk 1]
```diff
@@ -57,6 +57,7 @@ func TestCreateSpace(t *testing.T) {
 			Exists().
 			HasStatusTargetCluster("member-1").
 			HasConditions(spacetest.Provisioning()).
+			HasStateLabel("cluster-assigned").
```
Comment (xcoulon) [inline controllers/space/space_controller_test.go:60]: can we use a constant for `cluster-assigned`?

File: controllers/space/space_controller_test.go [hunk 2]
```diff
@@ -130,6 +133,7 @@ func TestCreateSpace(t *testing.T) {
 			assert.False(t, res.Requeue)
 			spacetest.AssertThatSpace(t, s.Namespace, s.Name, hostClient).
 				HasNoStatusTargetCluster().
+				HasStateLabel("pending").
```
Comment (xcoulon) [inline controllers/space/space_controller_test.go:136]: same here, can we use a constant for `pending`?

**Review by xcoulon — APPROVED** [review]
looks good, just a minor request/comment

File: controllers/space/space_controller_test.go (hunk 1 above)
Comment (MatousJobanek) [reply controllers/space/space_controller_test.go:60]: I didn't use the constant intentionally - so we know when the content has changed (for example by mistake)

File: controllers/space/space_controller.go [hunk 3]
```diff
@@ -345,6 +351,24 @@ func (r *Reconciler) deleteNSTemplateSetFromCluster(logger logr.Logger, space *t
...
+		// skipping
+		return nil
+	}
+	if space.Labels == nil {
+		space.Labels = map[string]string{}
+	}
+	space.Labels[toolchainv1alpha1.SpaceStateLabelKey] = state
+	if err := r.Client.Update(context.TODO(), space); err != nil {
+		return r.setStatusProvisioningFailed(logger, space, errs.Wrapf(err,
+			"unable to update state label at UserSignup resource"))
```
Comment (rajivnathan) [inline controllers/space/space_controller.go:366]:
```suggestion
			"unable to update state label at Space resource"))
```

**Review by rajivnathan — APPROVED** [review]

File: controllers/space/space_controller.go (hunk 3 above)
Comment (MatousJobanek) [reply controllers/space/space_controller.go:366]: Copy-paste driven development :man_facepalming: :smile: 
Thanks :+1:

**Pull request merged**